dora/core/client/fs/src/test/java/alluxio/client/file/dora/WorkerLocationPolicyTest.java

```diff
@@ -0,0 +1,217 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.client.file.dora;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+import alluxio.Constants;
+import alluxio.client.block.BlockWorkerInfo;
+import alluxio.client.file.dora.WorkerLocationPolicy.ConsistentHashProvider;
+import alluxio.wire.TieredIdentity;
+import alluxio.wire.WorkerNetAddress;
+
+import com.google.common.collect.ImmutableList;
+import org.apache.commons.lang3.RandomStringUtils;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.NavigableMap;
+import java.util.Set;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+public class WorkerLocationPolicyTest {
+  private static final long WORKER_LIST_TTL_MS = 100;
+  private static final String OBJECT_KEY = "/path/to/object";
+  private static final int NUM_VIRTUAL_NODES = 100;
+
+  @Test
+  public void uninitializedThrowsException() {
+    ConsistentHashProvider provider = new ConsistentHashProvider(1, WORKER_LIST_TTL_MS);
+    Assert.assertThrows(IllegalStateException.class, () -> provider.get(OBJECT_KEY, 0));
+  }
+
+  @Test
+  public void concurrentInitialization() {
+    ConsistentHashProvider provider = new ConsistentHashProvider(1, WORKER_LIST_TTL_MS);
+    final int numThreads = Runtime.getRuntime().availableProcessors();
```

Comment:
Lol. I guess you don't need to depend on numCPUs here. Tests on github are using probably 1 or 2 CPUs? And that can give you unexpected results against running on you Mac. I guess it's fine to keep using maybe 4 or 8. The latch will ensure those threads are concurrent (maybe not allocated CPU but they are indeed running concurrently).

Reply:
changed to a fixed value of `16`.